### PR TITLE
Cancel LAC watch when longpoll LAC times out

### PR DESCRIPTION
**Pull request by ivankelly**
Otherwise the watch objects will accumulate and eventually cause an
OOM on the bookie, if LAC doesn't progress.

### CODE REVIEW

**Comment by ivankelly** [comment]
It's hard to test an OOM type thing in a unit test, but if you want to see the problem in action, run a local bookie and run the following against it.

https://gist.github.com/ivankelly/65c999a1a3a280eea7e5b27e0a5a361b

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/LongPollReadEntryProcessorV3.java [hunk 1]
```diff
@@ -163,9 +163,10 @@ private ReadResponse getLongPollReadResponse() {
                 }
                 synchronized (this) {
                     expirationTimerTask = requestTimer.newTimeout(timeout -> {
-                        // When the timeout expires just get whatever is the current
-                        // readLastConfirmed
-                        LongPollReadEntryProcessorV3.this.scheduleDeferredRead(true);
+                            requestProcessor.bookie.cancelWaitForLastAddConfirmedUpdate(ledgerId, this);
```
Comment (eolivelli) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/LongPollReadEntryProcessorV3.java]: this is the core of the fix, isn't it ?

Comment (ivankelly) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/LongPollReadEntryProcessorV3.java]: @eolivelli yes

**Comment by ivankelly** [comment]
@eolivelli regarding mockito test, what would you assert on?

**Comment by ivankelly** [comment]
Must have neglected to clean before doing full build. Tests compile now

**Comment by eolivelli** [comment]
@ivankelly 
I was talking about ensuring that `LongPollReadEntryProcessorV3.java` actually calls `requestProcessor.bookie.cancelWaitForLastAddConfirmedUpdate`
which is the real fix.

This way we want drop that code in the future by mistake

**Comment by ivankelly** [comment]
@eolivelli Added test,

**Pull request merged**